.github/workflows/ci.yml

```diff
@@ -113,6 +117,24 @@ jobs:
       - name: Run 'make ${{ matrix.make-target }}'
         run: make ${{ matrix.make-target }}
 
+      - name: Analyse code coverage
+        if: matrix.code-coverage
+        uses: insightsengineering/coverage-action@v3.0.1
+        with:
+          path: src/riscv/cobertura.xml
+          threshold: 90
+          fail: true
+          publish: true
+          diff: true
+          diff-branch: main
+          diff-storage: _xml_coverage_reports
+          coverage-summary-title: "Code Coverage Summary"
+          uncovered-statements-increase-failure: true
```

Comment:
slightly worried this would be too strict, but can  see what happens once merged